pkg/controller/route53resolver/resolverrule/hooks.go

```diff
@@ -73,3 +76,20 @@ func preUpdate(_ context.Context, cr *v1alpha1.ResolverRule, obj *svcsdk.UpdateR
 	obj.ResolverRuleId = aws.String(meta.GetExternalName(cr))
 	return nil
 }
+
+func postObserve(_ context.Context, cr *svcapitypes.ResolverRule, obj *svcsdk.GetResolverRuleOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch aws.StringValue(obj.ResolverRule.Status) {
+	case string(svcapitypes.ResolverRuleStatus_SDK_COMPLETE):
+		cr.SetConditions(xpv1.Available())
+	case string(svcapitypes.ResolverRuleStatus_SDK_UPDATING), string(svcapitypes.ResolverRuleStatus_SDK_FAILED):
```

Comment:
Are we sure the resource is not available during update?

Reply:
yes we have cases if we change eg. the targetIPs